internal/images/image.go

```diff
@@ -0,0 +1,86 @@
+//
+// Copyright (c) 2019-2020 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package images
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+
+	"github.com/eclipse/che-plugin-broker/model"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var log = logf.Log.WithName("container-images")
+
+var envRegexp = regexp.MustCompile(`\${(.*)}`)
+
+const (
+	webTerminalToolingImageEnvVar  = "IMAGE_web_terminal_tooling"
```

Comment:
Why we have mixed upper and lower case here? Would not be it a bit consistent to use upper case only, like `IMAGE_WEB_TERMINAL_TOOLING`, or there is some requirement from OLM side?

Reply:
It's the style chosen by the CRW operator, and I feel like it reads a little better. Reading all-caps is tiring.

Reply:
> Environment variable names used by the utilities in the Shell and Utilities volume of IEEE Std 1003.1-2001 consist solely of uppercase letters, digits, and the '_' (underscore) from the characters defined in Portable Character Set and do not begin with a digit. Other characters may be permitted by an implementation; applications shall tolerate the presence of such names. ***Uppercase and lowercase letters shall retain their unique identities and shall not be folded together***. The name space of environment variable names containing lowercase letters is reserved for applications. Applications can define any environment variables with names from this name space without modifying the behavior of the standard utilities.

from https://pubs.opengroup.org/onlinepubs/000095399/basedefs/xbd_chap08.html

Google's shell convention declares upper case... https://google.github.io/styleguide/shellguide.html#s7.3-constants-and-environment-variable-names

I would follow a common convention that introducing our own but related images seem to be untypical )
There is a first place where I found `IMAGE_my_image` format https://docs.google.com/document/d/1n4MJgVk8g254s_X6oOgzC1z44RcJur3g2WaMuriv1fg/edit#
Design proposal uses lower case for related images names but it's not env vars: https://github.com/operator-framework/operator-lifecycle-manager/blob/release-4.3/doc/contributors/design-proposals/related-images.md
There is knative operator that uses: mixed upper and lower case, `-`, `_`, `__` for related image name and the same for the corresponding env vars
https://github.com/openshift-knative/serverless-operator/blob/master/olm-catalog/serverless-operator/1.8.0/serverless-operator.v1.8.0.clusterserviceversion.yaml

Seems CRW is going to rename that env vars from `IMAGE__*` to `RELATED_IMAGE_*` https://github.com/eclipse/che-operator/pull/274/files#diff-e7334c795c728921b138e6dde55e412dR48

The quick research I've done makes me even more confused. So, I'm OK with any option :-D

Reply:
I would add that most conventions (e.g.  IEEE Std 1003.1-2001) are specific to use in shells and shell utilities. The overall spec for env vars leaves it open to any character in the printable character set, and there's technically nothing in the spec preventing mixing case. I think it's a lot easier to parse the `RELATED_IMAGES_` prefix when case is mixed. 

```
RELATED_IMAGE_plugin_redhat_developer_web_terminal_4_5_0
RELATED_IMAGE_plugin_redhat_developer_web_terminal_nightly
RELATED_IMAGE_plugin_redhat_developer_web_terminal_dev_4_5_0
RELATED_IMAGE_plugin_redhat_developer_web_terminal_dev_nightly
RELATED_IMAGE_plugin_eclipse_cloud_shell_nightly
RELATED_IMAGE_web_terminal_tooling
RELATED_IMAGE_openshift_oauth_proxy
```
vs
```
RELATED_IMAGE_PLUGIN_REDHAT_DEVELOPER_WEB_TERMINAL_4_5_0
RELATED_IMAGE_PLUGIN_REDHAT_DEVELOPER_WEB_TERMINAL_NIGHTLY
RELATED_IMAGE_PLUGIN_REDHAT_DEVELOPER_WEB_TERMINAL_DEV_4_5_0
RELATED_IMAGE_PLUGIN_REDHAT_DEVELOPER_WEB_TERMINAL_DEV_NIGHTLY
RELATED_IMAGE_PLUGIN_ECLIPSE_CLOUD_SHELL_NIGHTLY
RELATED_IMAGE_WEB_TERMINAL_TOOLING
RELATED_IMAGE_OPENSHIFT_OAUTH_PROXY
```

Using this format also separates them from actual environment variables users might have to think about.

Reply:
> I think it's a lot easier to parse the RELATED_IMAGES_ prefix when case is mixed.

Agree